WordPress/Classes/Services/AccountSettingsService.swift

```diff
@@ -159,18 +159,18 @@ class AccountSettingsService {
         let request = NSFetchRequest(entityName: ManagedAccountSettings.entityName)
         request.predicate = NSPredicate(format: "account.userID = %d", userID)
         request.fetchLimit = 1
-        let results = (try? context.executeFetchRequest(request) as! [ManagedAccountSettings]) ?? []
+        let results = (try? context.executeFetchRequest(request)) as? [ManagedAccountSettings] ?? []
         return results.first
     }
 
     private func createAccountSettings(userID: Int, settings: AccountSettings) {
         let accountService = AccountService(managedObjectContext: context)
-        guard let account = accountService.findAccountWithUserID(userID) else {
+        guard let account = accountService.findAccountWithUserID(userID),
+        let managedSettings = NSEntityDescription.insertNewObjectForEntityForName(ManagedAccountSettings.entityName, inManagedObjectContext: context) as? ManagedAccountSettings else {
```

Comment:
The error message shown inside the guard only really makes sense if the first part of the guard fails.

I'd recommend a separate guard in this case for the second condition.